README.md

```diff
@@ -22,7 +22,82 @@ Guidance on onboarding samples to docs.microsoft.com/samples: https://review.doc
 Taxonomies for products and languages: https://review.docs.microsoft.com/new-hope/information-architecture/metadata/taxonomies?branch=master
 -->
 
-Service Fabric can leverage the Managed Identity for Azure Active Directory (AD) to manage credentials in your application. These code examples show how to do so.
+This walkthrough steps through the process of deploying a service fabric cluster to Azure with managed identity enabled, and then deploying an application that uses managed identity to that cluster.
+
+## Prerequisites
+
+- [You will need Powershell and the Az Powershell library to run this sample.](https://docs.microsoft.com/en-us/powershell/azure/install-az-ps)
+- You should have an Azure subscription and resource group you can write to. There is no way to test this sample locally.
+
+## Overview
+
+### Create a User-Assigned Managed Identity and Key Vault
+
+1. [Create a user assigned managed identity](https://docs.microsoft.com/en-us/azure/active-directory/managed-identities-azure-resources/how-to-manage-ua-identity-portal)
+    - Make sure you create it in the region you would like your cluster to be in
+2. Create a Key Vault and necessary cluster server and client certificates ([Below](###-step-2-create-a-key-vault-and-necessary-cluster-server-and-client-certificates))
+
+### Deploy a cluster
+
+3. Deploy a managed-identity-enabled cluster using an ARM template ([Below](###-step-3-deploy-a-managed-identity-enabled-cluster-using-an-arm-template
+))
+
+### Create an application
+
+4. Create a storage account to host your application package and upload it ([Below](###-step-4-create-a-storage-account-to-host-your-application-package-and-upload-it))
+5. Deploy an application
+    a. Deploy an application with user-assigned managed identity
+    b. Deploy code in a container with system-assigned managed identity
+
+## Walkthrough
+
+### Step 2: Create a Key Vault and necessary cluster server and client certificates
+<!---
+TODO: Provide an ARM template and deploy script to do this automatically
+--->
+#### Create Key Vault
+1. Under your resource group, click `+ Add`, choose `Key Vault`, and click `Create`
+2. Under `Basics`, fill out parameters, putting your keyvault in the same region that you would like your cluster to be in
+3. Under `Access Policy`, check `Azure Virtual Machines for Deployment`, this is what allows ARM to access to keyvault to get the cluster certificate

```

Comment:
".. this allows the provisioning of the cluster certificate during the VMSS deployment."